fix(api): fix incorrect path handling in Langfuse integration

chenguowei opened the pull request:
Solve the problem of langfuse configuration error   resolves #22757 

> [!IMPORTANT]
>
> 1. Make sure you have read our [contribution guidelines](https://github.com/langgenius/dify/blob/main/CONTRIBUTING.md)
> 2. Ensure there is an associated issue and you have been assigned to it
> 3. Use the correct syntax to link this PR: `Fixes #<issue number>`.

## Summary

<!-- Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change. -->

## Screenshots

| Before | After |
|--------|-------|
| ...    | ...   |

## Checklist

- [x] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)
- [x] I understand that this PR may be closed in case there was no previous discussion or issues. (This doesn't apply to typos!)
- [x] I've added a test for each change that was introduced, and I tried as much as possible to make a single atomic change.
- [x] I've updated the documentation accordingly.
- [x] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods

chenguowei committed:
Remove the truncation logic in the validate process

Nov1c444 commented:
Maybe It's the better way to fix the issue. 
change the `validate_endpoint_url` method in `config_entity.py` 
<img width="705" height="599" alt="image" src="https://github.com/user-attachments/assets/142b8aaf-895d-45bf-8e8f-921358401807" />

chenguowei commented:
The validate_url function itself only verifies whether the url is correct and should not change the original url. This is the definition of a validation function.

chenguowei committed:
fix validate_url test

QuantumGhost commented:
@chenguowei I agree with your principle that `validate_url` should focus solely on URL validation without modifying the input. However, I'm concerned about the potential impact of this change on the existing codebase.

The `validate_url` function is used across multiple locations in the codebase, and changing its behavior from returning a normalized URL to returning the original URL could introduce breaking changes in other components that depend on the current normalization behavior.

I suggest a more conservative approach:

1. Introduce `validate_url_with_path` for cases where path preservation is needed (like Langfuse configuration)
2. Mark `validate_url` as deprecated 
3. Gradually migrate existing usages to the appropriate function based on their requirements.

QuantumGhost committed:
docs(api): document the behavior of validate_url function

- Add a note about its behavior of not retaining the `path` component
  in its return value.
- Mark the function as deprecated (as in most situations,
  we need to retain the `path` component)
- Revert changes about `validate_url` and its test cases.

QuantumGhost committed:
fix(api): fix incorrect url validation in BaseTracingConfig

The validation function should keep the original path instead
of returning a modified result.

QuantumGhost committed:
refactor(api): decouple LangfuseConfig from BaseTracingConfig

Call `validate_url_with_path` directly in `LangfuseConfig`, instead of
calling `cls.validate_endpoint_url`. (The later introduces
a tight coupling between `LangfuseConfig` and `BaseTracingConfig`)

QuantumGhost committed:
chore(api): Making LangfuseConfig inherit from BaseTracingConfig

The `BaseTraceInstance` requires the configuration to be a subclass of
`BaseTracingConfig`.

QuantumGhost committed:
test(api): ensure `LangfuseConfig` correctly handles host with path component

Hammad663 commented:
Not interested in your artifactsi

chenguowei reviewed (commented):
by modifying case by case in this way, I understand that other subsequent modifications may also need to be made successively
